bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -554,6 +584,9 @@ func (r *KubeadmConfigReconciler) joinWorker(ctx context.Context, scope *Scope)
 	if res, err := r.reconcileDiscovery(ctx, scope.Cluster, scope.Config, certificates); err != nil {
 		return ctrl.Result{}, err
 	} else if !res.IsZero() {
+		if res.RequeueAfter == 0 {
```

Comment:
This and the below code line change were IMO missing cases where the reconciler returned that no requeueing is needed, but we need that while nodes haven't joined since the token may expire. It's covered in a changed test, so please double-check if that makes sense.

Reply:
I don't think this change is necessary.
reconcileDiscovery takes care of creating the token, it is not linked to the refresh token process